WalletWasabi.Fluent/ViewModels/Wallets/Send/SendFeeViewModel.cs

```diff
@@ -74,19 +101,33 @@ protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disp
 
 		Observable
 			.FromEventPattern(feeProvider, nameof(feeProvider.AllFeeEstimateChanged))
-			.Select(x => (x.EventArgs as AllFeeEstimate)!.Estimations)
+			.Select(_ =>
+			{
+				TransactionFeeHelper.TryGetFeeEstimates(_wallet, out var estimates);
+				return estimates;
+			})
+			.WhereNotNull()
 			.ObserveOn(RxApp.MainThreadScheduler)
-			.Subscribe(estimations => FeeChart.UpdateFeeEstimates(TransactionFeeHelper.GetFeeEstimates(_wallet), _transactionInfo.MaximumPossibleFeeRate))
+			.Subscribe(estimations => FeeChart.UpdateFeeEstimates(estimations, _transactionInfo.MaximumPossibleFeeRate))
 			.DisposeWith(disposables);
 
 		RxApp.MainThreadScheduler.Schedule(async () =>
 		{
-			while (feeProvider.AllFeeEstimate is null)
+			Dictionary<int, int> feeEstimates;
+			using var cancelTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(15));
+
+			try
+			{
+				feeEstimates = await TransactionFeeHelper.GetFeeEstimatesWhenReadyAsync(_wallet, cancelTokenSource.Token);
+			}
+			catch (Exception ex)
 			{
-				await Task.Delay(100);
+				Logger.LogInfo(ex);
+				await FeeEstimationsAreNotAvailableAsync();
+				return;
 			}
```

Comment:
Most of the PR is refactoring, the actual logic is here, start the review from this part.